sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -305,17 +328,23 @@
      * been cancelled. The completed operation returns a List of {@link FormPage}.
      * @throws FormRecognizerException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
      * an {@link OperationStatus#FAILED}.
-     * @throws NullPointerException If {@code recognizeOptions} is {@code null}.
+     * @throws NullPointerException If {@code form} is null.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PollerFlux<OperationResult, List<FormPage>> beginRecognizeContent(Flux<ByteBuffer> form, long length,
         RecognizeOptions recognizeOptions) {
         try {
+            Objects.requireNonNull(form, "'form' is required and cannot be null.");
+
             recognizeOptions = getRecognizeOptionsProperties(recognizeOptions);
             return new PollerFlux<>(
                 recognizeOptions.getPollInterval(),
-                contentStreamActivationOperation(form, length, recognizeOptions.getContentType()),
-                extractContentPollOperation(),
+                streamActivationOperation(
+                    contentType -> service.analyzeLayoutAsyncWithResponseAsync(contentType, form, length)
+                        .map(response -> new OperationResult(
+                            parseModelId(response.getDeserializedHeaders().getOperationLocation()))),
+                    form, recognizeOptions.getContentType()),
+                pollingOperation(service::getAnalyzeLayoutResultWithResponseAsync),
                 (activationResponse, context) ->
                     monoError(logger, new RuntimeException("Cancellation is not supported")),
                 fetchExtractContentResult());
```

Comment:
Why not update the fetching operation here?

Reply:
It returns `Mono<List<FormPage>>` other fetching returns `Mono<List<RecognizedForm>>`

and FormPage and RecognizedForm has no common parent.

Reply:
So you just grouped the receipt and form API ?

Reply:
There are two places are using fetchExtractContentResult() method: `beginRecognizeContentFromUrl` and `beginRecognizeContent`. I leave it unchange in both place since there is no benefit of doing so.

Reply:
Updated to use the updated fetchingOperation(). Bring the complexity to upper level but it makes code more clean even we added more line of code.